File: Modules/ServerTypes/CustomerManagement/CustomerColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CustomerRow } from "./CustomerRow";

export interface CustomerColumns {
    CustomerId: Column<CustomerRow>;
    CustomerName: Column<CustomerRow>;
    ContactName: Column<CustomerRow>;
    Email: Column<CustomerRow>;
    Phone: Column<CustomerRow>;
    City: Column<CustomerRow>;
    District: Column<CustomerRow>;
    Country: Column<CustomerRow>;
}

export class CustomerColumns extends ColumnsBase<CustomerRow> {
    static readonly columnsKey = 'CustomerManagement.Customer';
    static readonly Fields = fieldsProxy<CustomerColumns>();
}

File: Modules/CustomerManagement/Customer/CustomerColumns.cs
namespace MapPin.CustomerManagement.Forms;

[ColumnsScript("CustomerManagement.Customer")]
[BasedOnRow(typeof(CustomerRow), CheckNames = true)]
public class CustomerColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int CustomerId { get; set; }

    [EditLink, SortOrder(1)]
    public string CustomerName { get; set; }

    public string ContactName { get; set; }

    public string Email { get; set; }

    public string Phone { get; set; }

    public string City { get; set; }

    public string District { get; set; }

    public string Country { get; set; }
}
